From f187c111c62dd7c06b6750c0ffc54e5bf9f545fe Mon Sep 17 00:00:00 2001
From: Alexandre Ghiti <alexandre.ghiti@canonical.com>
Date: Tue, 1 Feb 2022 11:36:29 +0100
Subject: [PATCH] UBUNTU: Add SPL CONFIG to visionfive defconfig

This allows SPL from the flash to load openSBI and u-boot proper from
the sdcard.

Signed-off-by: Alexandre Ghiti <alexandre.ghiti@canonical.com>
---
 configs/starfive_jh7100_visionfive_smode_defconfig | 8 ++++++++
 1 file changed, 8 insertions(+)

--- a/configs/starfive_jh7100_visionfive_smode_defconfig
+++ b/configs/starfive_jh7100_visionfive_smode_defconfig
@@ -1,6 +1,7 @@
 CONFIG_RISCV=y
 CONFIG_SYS_MALLOC_LEN=0x800000
 CONFIG_SYS_MALLOC_F_LEN=0x3000
+CONFIG_SPL_GPIO=y
 CONFIG_NR_DRAM_BANKS=1
 CONFIG_SYS_MEMTEST_START=0x80000000
 CONFIG_SYS_MEMTEST_END=0x200000000
@@ -8,6 +9,8 @@
 CONFIG_ENV_SECT_SIZE=0x1000
 # CONFIG_DM_GPIO is not set
 CONFIG_DEFAULT_DEVICE_TREE="jh7100-visionfive"
+CONFIG_SPL_MMC=y
+CONFIG_SPL=y
 CONFIG_IDENT_STRING="StarFive"
 CONFIG_SYS_CLK_FREQ=1000000000
 CONFIG_TARGET_STARFIVE_JH7100=y
@@ -21,6 +24,7 @@
 CONFIG_SYS_BOOT_GET_KBD=y
 CONFIG_SYS_LOAD_ADDR=0x80200000
 CONFIG_FIT=y
+CONFIG_SPL_LOAD_FIT_ADDRESS=0x84000000
 CONFIG_CHROMEOS=y
 CONFIG_SHOW_BOOT_PROGRESS=y
 CONFIG_QSPI_BOOT=y
@@ -41,6 +45,10 @@
 CONFIG_DISPLAY_BOARDINFO=y
 CONFIG_MISC_INIT_R=y
 CONFIG_ID_EEPROM=y
+CONFIG_SPL_SEPARATE_BSS=y
+CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_USE_SECTOR=y
+CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_SECTOR=0x800
+CONFIG_SPL_CACHE=y
 CONFIG_SYS_PROMPT="VisionFive #"
 CONFIG_CMD_CONFIG=y
 CONFIG_CMD_LICENSE=y
